core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -285,16 +285,12 @@ private[spark] object Utils extends Logging {
    */
   def createDirectory(dir: File): Boolean = {
     try {
-      // This sporadically fails - not sure why ... !dir.exists() && !dir.mkdirs()
-      // So attempting to create and then check if directory was created or not.
-      dir.mkdirs()
-      if ( !dir.exists() || !dir.isDirectory) {
-        logError(s"Failed to create directory " + dir)
-      }
-      dir.isDirectory
+      // SPARK-35907 Instead of File#mkdirs, Files#createDirectories is expected.
```

Comment:
Shall we remove this line? Since `mkdirs` is removed already, we may look weird after this PR.

Reply:
> Shall we remove this line? Since `mkdirs` is removed already, we may look weird after this PR.

OK, I will provide more appropriate comments.